src/functional/iterator/embedded.py

```diff
@@ -586,11 +606,28 @@ def shape(self):
         return self.array().shape
 
 
+def _is_tuple_axis(axis: Axis):
+    return axis is None
+
+
 def get_ordered_indices(
-    axes: Iterable[Axis], pos: Mapping[str, FieldIndex]
+    axes: Iterable[Axis], pos: Mapping[str, FieldIndex | SparsePositionEntry]
 ) -> tuple[FieldIndex, ...]:
-    assert all(axis.value in [*pos.keys()] for axis in axes)
-    return tuple(pos[axis.value] for axis in axes)
+    res = list[FieldIndex]()
+    pos = deepcopy(pos)
+    for axis in axes:
+        if _is_tuple_axis(axis):
+            res.append(slice(None))
+        else:
+            assert axis is not None
```

Comment:
redundant? `_is_tuple_axis` is checking exactly the same

Reply:
With `if _is_tuple_axis(axis):` i wanted to highlight the tuple axis check (instead of using `if isinstance(axis, None)`), however now in the else mypy doesn't understand that I excluded None already.

Reply:
I think the `assert` will finally become unnecessary after PR #841 gets merged, since the `_is_tuple_axis` was converted to a TypeGuard, so I recommend to merge that one first and then update this PR.